pysilcam/tracking/track.py

```diff
@@ -0,0 +1,417 @@
+        except:
+            try:
+                times = [f.replace(datapath + '/D', '').replace('.silc_mono', '') for f in files]
+                times = pd.to_datetime(times)
+            except:
+                pass
+
+    t1 = times[0]
+
+    dt = np.abs(times - t1)
+
+    secs = (dt.components.hours * 3600) + (dt.components.minutes * 60) + dt.components.seconds + (
+            dt.components.milliseconds / 1000)
+    image_number = np.arange(0, len(secs))
+
+    base = 10
+    start = 0
+    stop = max(secs)
+    sample_secs = np.logspace(start, np.log(stop) / np.log(base), num=approx_files, base=base)
+    sample_secs = sample_secs[sample_secs < max(secs)]
+    f = interpolate.interp1d(secs, image_number, kind='nearest')
+    sample_idx = f(sample_secs)
+    sample_idx_ = []
+    for i in sample_idx:
+        sample_idx_.append(int(i))
+    sample_idx = np.unique(sample_idx_)
+
+    len(sample_idx)
+
+    files = np.array(np.copy(files))
+    resampled_files = sorted(list(files[sample_idx]))
+
+    print('  ', str(len(resampled_files)), 'files from', str(len(files)))
+
+    return resampled_files
+
+
+def plot_single(datapath):
+    outputpath = make_output_path(datapath)
+    csv_file = (outputpath + '.csv')
+    print('CSV file:', csv_file)
+
+    data = pd.read_csv(csv_file)
+    print(data.columns)
+
+    f, a = plt.subplots(2, 2, figsize=(12, 12))
+
+    plt.sca(a[0, 0])
+    plt.plot(pd.to_datetime(data['Time']), data['ECD [mm]'], '.', color='0.8', alpha=0.1)
+    plt.ylabel('ECD [mm]')
+
+    plt.sca(a[1, 0])
+    plt.plot(pd.to_datetime(data['Time']), data['Speed [cm/s]'], '.', color='0.8', alpha=0.1)
+    plt.ylabel('Speed [cm/s]')
+
+    plt.sca(a[0, 1])
+    plt.plot(data['ECD [mm]'], data['Ws [cm/s]'], '.', color='0.8', alpha=0.1)
+    plt.xlabel('ECD [mm]')
+    plt.ylabel('Ws [cm/s]')
+
+    plt.sca(a[1, 1])
+    plt.plot(data['ECD [mm]'], data['Speed [cm/s]'], '.', color='0.8', alpha=0.1)
+    plt.xlabel('ECD [mm]')
+    plt.ylabel('Speed [cm/s]')
+
+    plt.show()
+
+
+def load_and_process(tracksfile, PIX_SIZE,
+                     minlength=0, maxlength=1e6, track_length_limit=15):
+    data = pd.read_hdf(tracksfile, 'Tracking/data')
+    tracks = post_process(data, PIX_SIZE,
+                          track_length_limit=track_length_limit,
+                          minlength=minlength, maxlength=maxlength)
+    return data, tracks
+
+
+def track_process(configfile, datapath, offset=0):
+    '''
+    This takes a silcam config file (and uses information within the [Tracking] section to track particles in images in
+    datapath. This will produce a -TRACKS.h5 file in the [General]datafile location (in a similar manner to normal
+    silcam processing).
+
+    Functions like pysilcam.tracking.track.make_boxplot can make plots with these -TRACK.h5 files.
+
+    :param configfile:
+    :param datapath:
+    :param offset:
+    :return:
+    '''
+    settings = PySilcamSettings(configfile)
+
+    sctr = Tracker()
```

Comment:
F405: 'Tracker' may be undefined, or defined from star imports: pysilcam.tracking.silcamtracker  <!-- Automatically posted by Lintly -->